src/firmware/app/control/trajectory_planner.h

```diff
@@ -152,27 +124,87 @@ app_trajectory_planner_interpolateConstantPeriodPositionTrajectory(
     PositionTrajectory_t* variable_time_trajectory, const float interpolation_period);
 
 /**
- * Generates the X/Y points for each position on the constant arc length trajectory
+ * Builds the forwards continuous linear velocity profile for the given parameters.
```

Comment:
At this point this is a pretty massive header file. What do people think about creating `trajectory_planner_impl.h` which would have all the "private" functions that we still want to expose for testing. We could then create a `trajectory_planner_impl` bazel target that is only visible from the `trajectory_planner` and `trajectory_planner_test` targets to ensure no one depends on the "private" functions.

Reply:
Can we break it up into linear_trajectory_planner and angular_trajectory_planner as sub files or is it too well integrated? (It would be bad if it were)

Reply:
The vast majority of the code is shared, so I imagine not, but @EvanMorcom would know better.

Reply:
You can break up the time-optimal generation of the linear and angular trajectories into their own files (but a lot of the code is shared)

The rebalancing using both profiles obviously can't be separated.

Reply:
Outdates with new refactor

Reply:
After in-person discussion we agreed that it's still worth doing this.

Reply:
Moved helper functions to _impl files